Xamarin.Forms.Platform.Android/FastRenderers/LabelRenderer.cs

```diff
@@ -324,7 +324,11 @@ void UpdateLineBreakMode()
 		void UpdateMaxLines()
 		{
 			SetSingleLine(Element.MaxLines == 1);
-			SetMaxLines(Element.MaxLines > 0 ? Element.MaxLines : 1);
+
+			if (Element.MaxLines == (int)Label.MaxLinesProperty.DefaultValue)
+				SetMaxLines(int.MaxValue);
+			else
+				SetMaxLines(Element.MaxLines > 0 ? Element.MaxLines : 1);
```

Comment:
Here, we could pass Element.MaxLines without the predicate check. I'm curious to know how Android interprets 0.